core/src/services/config.js

```diff
@@ -1,85 +1,96 @@
-import { isFunction, isPromise } from '../utilities/helpers.js';
+import { getConfigValueFromObjectAsync } from '../utilities/async-helpers';
+import { getConfigValueFromObject } from '../utilities/helpers';
 
-(function() {
-  window.Luigi = {
-    setConfig: configInput => {
-      window.Luigi.config = configInput;
-    },
-    getConfig: () => {
-      return window.Luigi.config;
-    }
-  };
-})();
-/*
- * Gets value of the given property on the given object.
- */
-export const getConfigValueFromObject = (object, property) => {
-  let propIndex = 0;
-  let nextValue = object;
-  const propertyPath = property.split('.');
-  while (nextValue && propIndex < propertyPath.length) {
-    nextValue = nextValue[propertyPath[propIndex++]];
+class LuigiConfigManager {
+  constructor() {
+    this.configReadyTimeout = {
+      valueMs: 2000,
+      id: undefined
+    };
+    this.config;
+
+    this.configReadyCallback = function() {};
```

Comment:
I think here's the same story as with `this.config;` above. It doesn't change anything if you remove this line.

Reply:
This initial value is meant to be able to call this.configReadyCallback without the need of an `if` block to prove if it is defined or not.